File: src/matlab/risk_analysis.m
function var = calculate_var(returns, confidence_level)
    % Calculates Value at Risk (VaR)
    sorted_returns = sort(returns);
    index = round((1 - confidence_level) * length(sorted_returns));
    var = sorted_returns(index);
end

function cvar = calculate_cvar(returns, confidence_level)
    % Calculates Conditional VaR (CVaR)
    var = calculate_var(returns, confidence_level);
    cvar = mean(returns(returns <= var));
end

% Example usage
% returns = randn(1000, 1) * 0.02;
% var = calculate_var(returns, 0.95);
% cvar = calculate_cvar(returns, 0.95);
% fprintf('VaR: %.4f, CVaR: %.4f\n', var, cvar);

File: src/python/risk_analysis.py
import numpy as np

def calculate_var(returns, confidence_level=0.95):
    """
    Calculates Value at Risk (VaR) for the given confidence level.
    """
    var = np.percentile(returns, (1 - confidence_level) * 100)
    print(f"Value at Risk (VaR) at {confidence_level} confidence level: {var}")
    return var

def calculate_cvar(returns, confidence_level=0.95):
    """
    Calculates Conditional VaR (CVaR) for the given confidence level.
    """
    var = calculate_var(returns, confidence_level)
    cvar = returns[returns <= var].mean()
    print(f"Conditional VaR (CVaR) at {confidence_level} confidence level: {cvar}")
    return cvar
